### PR TITLE
[SYCL-MLIR][NFC]: Code cleanup CGCall.cc and clang-mlir.h

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Comment by victor-eds** [comment]
I'm guessing this should be merged after #7326

**Comment by etiotto** [comment]
> I'm guessing this should be merged after #7326

No, the commits were messed up, but I rebased my changes and now only 3 files show diffs (which is correct).

File: polygeist/tools/cgeist/Lib/CGCall.cc [hunk 1]
```diff
@@ -462,844 +461,539 @@ ValueCategory MLIRScanner::VisitCallExpr(clang::CallExpr *expr) {
...
-    if (auto *lhs = dyn_cast<CXXTypeidExpr>(oc->getImplicitObjectArgument())) {
-      expr->getCallee()->dump();
-      if (auto *ic = dyn_cast<MemberExpr>(expr->getCallee()))
-        if (auto *sr = dyn_cast<NamedDecl>(ic->getMemberDecl())) {
-          if (sr->getIdentifier() && sr->getName() == "name") {
-            QualType LT = lhs->isTypeOperand()
-                              ? lhs->getTypeOperand(Glob.getCGM().getContext())
-                              : lhs->getExprOperand()->getType();
+  if (auto *Oc = dyn_cast<CXXMemberCallExpr>(Expr)) {
+    if (auto *Lhs = dyn_cast<CXXTypeidExpr>(Oc->getImplicitObjectArgument())) {
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Lib/CGCall.cc:500]:
```suggestion
    if (auto *LHS = dyn_cast<CXXTypeidExpr>(Oc->getImplicitObjectArgument())) {
```

File: polygeist/tools/cgeist/Lib/CGCall.cc [hunk 2]
```diff
@@ -1580,123 +1274,123 @@ MLIRScanner::EmitGPUCallExpr(clang::CallExpr *expr) {
       }
     }
 
-    auto createBlockIdOp = [&](gpu::Dimension str,
-                               mlir::Type mlirType) -> mlir::Value {
+    auto CreateBlockIdOp = [&](gpu::Dimension Str,
+                               mlir::Type MlirType) -> mlir::Value {
```
Comment (whitneywhtsang) [inline polygeist/tools/cgeist/Lib/CGCall.cc:1278]:
```suggestion
                               mlir::Type MLIRType) -> mlir::Value {
```

**Review by whitneywhtsang — APPROVED** [review]

**Pull request merged**